Hiddenvote

vzapo opened the pull request:
Added hidden vote command. This ticket can close #30 

Both timed and normal poll can now use hidden concept. The commands now requires an extra argument which defines the hidden status of the vote. 
On `tpoll`  timed poll command if the second argument is set to 'yes' it will act as an hidden timed poll. If it's set to 'no' it will act as a normal timed poll.
Same workflow for `poll` command just that it's the first argument not second.

meleu commented:
People are already used with the `!poll question option1 option2 optionN` notation.

Will they still be able to use such notation?

After a quick glance on the code, it seems like the `yes` or `no` as the first argument is mandatory. Am I right?

(BTW, sorry for not being able to review your other PRs. I'm kinda busy with other projects, but fortunately they're JS projects. Hope to be able to give more attention here soon.)

vzapo commented:
this still works the same ,the only thing that changed for both  commands are a new argument, which unfortunate it is mandatory. commands will now be as `tpoll time hidden question option1 option2 optionN` &  `poll hidden question option1 option2 optionN`.

meleu commented:
oh, changing the way the users are used to call the command is not something I would like to merge.

One solution I can think is making it like a classic UNIX command and add the **possibility** to use `--hidden` as the first argument (it would be **optional**) and if such option is present, the votes will be hidden.

The other solution would be to create brand new commands for hidden votes (`!hiddenpoll`, `!hpoll`, or something.)

vzapo commented:
Then the best would be to use a new command I would say. The command could be called `htpoll / hpoll` form hidden poll ?:)

meleu commented:
> The command could be called `htpoll / hpoll` form hidden poll ?:)

Looks good.
Probably the long form being the "official name" and short one being an alias.
Like `hiddenpoll` -> `hpoll`
and `hiddentimedpoll` -> `htpoll`

Thanks dude! :wink:

vzapo commented:
- reverted `poll` and `tpoll` commands
- added hidden timed poll command as `htpoll seconds hidden question option1 option2 optionN` and alias `hiddentimedpoll`
- changed messaged on `poll` command description(removed the 'usless' word from description 😄 )
